### PR TITLE
Avoid a phpcs:ignore and reformat option_sticky_posts()

### PR DESCRIPTION
**Pull request by Chouby**
The main purpose is to remove a `PHPCS:ignore WordPress.DB.PreparedSQL`.
I took the opportunity to reformat the code to remove some pyramid steps.

### CODE REVIEW

File: frontend/frontend-filters.php [hunk 1]
```diff
@@ -66,41 +66,47 @@ public function get_locale() {
...
 
-				foreach ( $relations as $relation ) {
-					$_posts[ $relation->term_taxonomy_id ][] = (int) $relation->object_id;
-				}
-				wp_cache_add( 'sticky_posts', $_posts, 'options' );
-			}
+		$languages = array();
+		foreach ( $this->model->get_languages_list() as $language ) {
+			$languages[] = $language->get_tax_prop( 'language', 'term_taxonomy_id' );
+		}
```
Comment (Hug0-Drelon) [inline frontend/frontend-filters.php:89]:
```suggestion
		$languages = array_map(
		    function ( $language ) {
		        return $language->get_tax_prop( 'language', 'term_taxonomy_id' );
		    },
		    $this->model->get_languages_list()
		)
```
Just suggesting a more readable version according to my taste 😅

File: frontend/frontend-filters.php [hunk 2]
```diff
@@ -66,41 +66,47 @@ public function get_locale() {
...
-			if ( empty( $_posts ) || ! is_array( $_posts ) || empty( $_posts[ $tt_id ] ) || ! is_array( $_posts[ $tt_id ] ) ) {
-				$posts = array_map( 'intval', $posts );
-				$posts = implode( ',', $posts );
-
-				$languages = array();
-				foreach ( $this->model->get_languages_list() as $language ) {
-					$languages[] = $language->get_tax_prop( 'language', 'term_taxonomy_id' );
-				}
+		// Do not filter sticky posts on REST requests as $this->curlang is *not* the 'lang' parameter set in the request.
+		if ( defined( 'REST_REQUEST' ) || empty( $this->curlang ) || empty( $posts ) ) {
```
Comment (Hug0-Drelon) [inline frontend/frontend-filters.php:75]: Is it old code that's never applied?
I mean, can `PLL_Frontent_Filters` be instantiated during a REST request?

**Review by Hug0-Drelon — CHANGES_REQUESTED** [review]

File: frontend/frontend-filters.php (hunk 1 above)
Comment (Chouby) [reply frontend/frontend-filters.php:89]: The opposite for me

File: frontend/frontend-filters.php (hunk 2 above)
Comment (Chouby) [reply frontend/frontend-filters.php:75]: Possible it's useless now. I prefer to keep as is as it doesn't harm (and don't want to spend time on it now).

File: frontend/frontend-filters.php [hunk 3]
```diff
@@ -66,41 +66,47 @@ public function get_locale() {
...
+					implode( ',', array_fill( 0, count( $posts ), '%d' ) ),
+					implode( ',', array_fill( 0, count( $languages ), '%d' ) )
+				),
+				array_merge( $posts, $languages )
+			)
+		);
+
+		$_posts = array_fill_keys( $languages, array() ); // Init with empty arrays.
+
+		foreach ( $relations as $relation ) {
```
[the remark this answers is not in the document]
Comment (Hug0-Drelon) [reply frontend/frontend-filters.php:104]: Indeed, thanks for the insight.

**Review by Hug0-Drelon — APPROVED** [review]

**Pull request merged**